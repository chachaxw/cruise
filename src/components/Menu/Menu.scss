@import '../../styles/_variables';

.cruise-menu {
    width: 250px;
    padding-top: 20px;
    background-color: $menuBackground;

    .cruise-menu-item {
        
        a {
            text-decoration: none;
            display: flex;
            color: #FFFFFF;
            font-size: 14px;
            font-weight: normal;
            padding: 13px 20px 12px;
            align-items: center;
        }

        &:hover a, .active {
            color: $cyanBlue;
            background-color: $menuHover;
        }
        
        .iconfont {
            margin-right: 15px;
        }
    }

}