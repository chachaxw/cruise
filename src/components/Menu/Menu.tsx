import * as React from 'react';
import { NavLink } from 'react-router-dom';
import './Menu.scss';

export type MenuItem = {
    to: string,
    icon: string,
    name: string,
}

interface InternalProps {
    menu: MenuItem[];
}

export default class Menu extends React.Component<InternalProps> {

    public renderMenuItem(props: MenuItem, index: number) {
        return (
            <li key={index} className="cruise-menu-item">
                <NavLink to={props.to} activeClassName="active">
                    <span className={`iconfont ${props.icon}`} ></span>
                    <span>{props.name}</span>
                </NavLink>
            </li>
        );
    }

    public render() {
        const { menu } = this.props;

        return (
            <div className="cruise-menu">
                <ul>
                    {menu && menu.length && menu.map((item: MenuItem, index: number) => (
                        this.renderMenuItem(item, index)
                    ))}
                </ul>
            </div>
        );
    }
}
