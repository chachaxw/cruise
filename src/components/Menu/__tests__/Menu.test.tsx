import * as React from 'react';
import { StaticRouter } from 'react-router-dom';
import TestRenderer from 'react-test-renderer';
import Menu from '../Menu';

it('renders with given prop menu', () => {
    const test = [
        { to: '/test1', icon: 'icon-test1', name: 'TEST1' },
        { to: '/test2', icon: 'icon-test2', name: 'TEST2' },
        { to: '/test3', icon: 'icon-test3', name: 'TEST3' },
        { to: '/test4', icon: 'icon-test4', name: 'TEST4' },
    ];

    const testRenderer = TestRenderer.create(
        <StaticRouter context={{}}>
            <Menu menu={test}></Menu>
        </StaticRouter>
    );

    expect(testRenderer.toJSON()).toMatchSnapshot();
});