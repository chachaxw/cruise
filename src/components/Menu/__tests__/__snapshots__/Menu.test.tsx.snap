// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with given prop menu 1`] = `
<div
  className="cruise-menu"
>
  <ul>
    <li
      className="cruise-menu-item"
    >
      <a
        aria-current={null}
        href="/test1"
        onClick={[Function]}
      >
        <span
          className="iconfont icon-test1"
        />
        <span>
          TEST1
        </span>
      </a>
    </li>
    <li
      className="cruise-menu-item"
    >
      <a
        aria-current={null}
        href="/test2"
        onClick={[Function]}
      >
        <span
          className="iconfont icon-test2"
        />
        <span>
          TEST2
        </span>
      </a>
    </li>
    <li
      className="cruise-menu-item"
    >
      <a
        aria-current={null}
        href="/test3"
        onClick={[Function]}
      >
        <span
          className="iconfont icon-test3"
        />
        <span>
          TEST3
        </span>
      </a>
    </li>
    <li
      className="cruise-menu-item"
    >
      <a
        aria-current={null}
        href="/test4"
        onClick={[Function]}
      >
        <span
          className="iconfont icon-test4"
        />
        <span>
          TEST4
        </span>
      </a>
    </li>
  </ul>
</div>
`;
